tests/PHPStan/Analyser/LegacyNodeScopeResolverTest.php

```diff
@@ -5129,35 +5129,35 @@ public function dataArrayFunctions(): array
 				'array_values($generalStringKeys)',
 			],
 			[
-				'array<int|non-empty-string, stdClass>',
+				'array<int|non-empty-string, stdClass>&nonEmpty',
 				'array_merge($stringOrIntegerKeys)',
 			],
 			[
-				'array<int|string, DateTimeImmutable|int>',
+				'array<int|string, DateTimeImmutable|int>&nonEmpty',
 				'array_merge($generalStringKeys, $generalDateTimeValues)',
 			],
 			[
-				'array<int|string, int|stdClass>',
+				'array<int|string, int|stdClass>&nonEmpty',
 				'array_merge($generalStringKeys, $stringOrIntegerKeys)',
 			],
 			[
-				'array<int|string, int|stdClass>',
+				'array<int|string, int|stdClass>&nonEmpty',
 				'array_merge($stringOrIntegerKeys, $generalStringKeys)',
 			],
 			[
-				'array<int|non-empty-string, \'foo\'|stdClass>',
+				'array<int|non-empty-string, \'foo\'|stdClass>&nonEmpty',
 				'array_merge($stringKeys, $stringOrIntegerKeys)',
 			],
 			[
-				'array<int|non-empty-string, \'foo\'|stdClass>',
+				'array<int|non-empty-string, \'foo\'|stdClass>&nonEmpty',
 				'array_merge($stringOrIntegerKeys, $stringKeys)',
 			],
 			[
-				'array<int|non-empty-string, 2|4|\'a\'|\'b\'|\'green\'|\'red\'|\'trapezoid\'>',
+				'array<int|non-empty-string, 2|4|\'a\'|\'b\'|\'green\'|\'red\'|\'trapezoid\'>&nonEmpty',
 				'array_merge(array("color" => "red", 2, 4), array("a", "b", "color" => "green", "shape" => "trapezoid", 4))',
 			],
 			[
-				'array<int|string, DateTimeImmutable|int>',
+				'array<int|string, DateTimeImmutable|int>&nonEmpty',
 				'array_merge(...[$generalStringKeys, $generalDateTimeValues])',
```

Comment:
Im am not confident, whether this is a expected change, or the impl is buggy for this case.